parkbenchcommon/broadcaster.py

```diff
@@ -0,0 +1,111 @@
+  substantial restructuring of the entire project.
+"""
+
+__all__ = ['BroadcasterIssueException',
+           'BroadcasterInitException',
+           'Broadcaster']
+
+import datetime
+import logging
+import os
+import stat
+import traceback
+from parkbenchcommon import daemonhelper
+from parkbenchcommon import ramdisk
+
+SPOOL_PATH = '/var/spool'
+RAMDISK_SIZE = '1M'
+# drwx--x---
+PROGRAM_DIR_MODE = stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR | stat.S_IXGRP
+# drwxr-x---
+BROADCAST_DIR_MODE = stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR | stat.S_IRGRP | stat.S_IXGRP
+
+class BroadcasterIssueException(Exception):
+    """This exception is raised when a Broadcaster object fails to issue a broadcast."""
+
+class BroadcasterInitException(Exception):
+    """This exception is raised when a Broadcaster object fails to initialize."""
+
+class Broadcaster(object):
+    """Provides the broadcasting component of a filesystem-based IPC mechanism."""
+
+    def __init__(self, program_name, broadcast_name, uid, gid):
+        """Initial configuration of the broadcast directory. This must be done as root. This
+        constructor mounts a ramdisk and creates any necessary spool directories with
+        proper permissions. Any failure to create directories will raise exceptions.
+
+        program_name: The name of the program issuing the broadcast.
+        broadcast_name: The name of the broadcast to issue.
+        uid: The UID of the calling program.
+        gid: The GID of the calling program.
+        """
+
+        self.logger = logging.getLogger(__name__)
+        self.logger.debug('Initializing broadcaster for broadcast %s from program %s.',
+                          broadcast_name, program_name)
+
+        self.program_name = program_name
+        self.broadcast_name = broadcast_name
+
+        program_path = os.path.join(SPOOL_PATH, program_name)
+        ramdisk_path = os.path.join(program_path, 'ramdisk')
+        self.ramdisk = ramdisk.Ramdisk(ramdisk_path)
+        self.broadcast_path = os.path.join(ramdisk_path, 'broadcast')
+
+        self.logger.debug('Creating broadcast directories for program %s.', program_name)
+        daemonhelper.create_directories(SPOOL_PATH, ramdisk_path, uid, gid, PROGRAM_DIR_MODE)
+
+        self.ramdisk.mount(RAMDISK_SIZE, uid, gid, PROGRAM_DIR_MODE)
+        daemonhelper.create_directories(
+            ramdisk_path, 'broadcast', uid, gid, BROADCAST_DIR_MODE)
+
+        self.logger.info('Broadcaster %s from program %s initialized.',
+                         broadcast_name, program_name)
+
+    def issue(self):
+        """Issues a new broadcast overriding any prior broadcast. Will raise an exception if
+        it fails.
+        """
+        self.logger.info(
+            'Issuing broadcast %s for program %s.', self.broadcast_name, self.program_name)
+        now = datetime.datetime.now().isoformat()
+        # A random-ish number is added to the filename to avoid filename collisions.
```

Comment:
I'm wondering if people will read "ish" and think we aren't using a secure random number generator.